src/components/userDasboard/components/DashboardDetails.jsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Link } from 'react-router-dom';
+
+const DashboardDetails = ({ numberOfOrgs, numberOfDictionary, organizations }) => {
+  const nameOfOrganizations = organizations.map(organization => organization.name).join(',');
+  const dictionary = numberOfDictionary < 2 ? 'dictionary' : 'dictionaries';
+  if (numberOfOrgs === 1) {
+    return (
+      <div className="user-data">
+        <p className="lead">
+          You belong to {nameOfOrganizations} organization. This can be changed via{' '}
+          <a href="https://qa.openconceptlab.org/" target="_blank" rel="noopener noreferrer">
+            the traditional OCL
+          </a>.
+        </p>
+        <p className="lead">
+          You can <Link to="dashboard/dictionaries">view all OpenMRS public dictionaries</Link>.
+        </p>
+      </div>
+    );
+  } else if (numberOfOrgs > 1) {
+    return (
+      <div className="user-data">
+        <p className="lead">
+          You belong to {numberOfOrgs} organizations:{' '}
+          {organizations.map(organization => (
+            <span className="d-block lead org-name text-capitalize" key={organization.id}>
+              <a
+                href={`https://qa.openconceptlab.org${organization.url}`}
+                target="_blank"
+                rel="noopener noreferrer"
+              >
+                <i className="fas fa-arrow-circle-right fa-fw" /> {organization.name}
+              </a>
+            </span>
+          ))}{' '}
+          This can be changed via{' '}
+          <a href="https://qa.openconceptlab.org/" target="_blank" rel="noopener noreferrer">
+            the traditional OCL
+          </a>.
+        </p>
+        <p className="lead">
+          You can <Link to="dashboard/dictionaries">view all OpenMRS public dictionaries</Link>.
+        </p>
+      </div>
+    );
+  }
+  return (
+    <div className="user-data">
+      <p className="lead">
+        You do not belong to any organization. This can be changed via{' '}
+        <a href="https://qa.openconceptlab.org/" target="_blank" rel="noopener noreferrer">
+          the traditional OCL
+        </a>.
+      </p>
+      <p className="lead">
+        You currently have {numberOfDictionary} personal {dictionary}. You can{' '}
+        <Link to="dashboard/dictionaries">view all OpenMRS public dictionaries</Link>.
+      </p>
+    </div>
+  );
+};
+
+DashboardDetails.propTypes = {
+  numberOfOrgs: PropTypes.number.isRequired,
+  numberOfDictionary: PropTypes.number.isRequired,
+  organizations: PropTypes.array.isRequired,
+};
+
```

Comment:
This component is rendering three different views based on some condition, don't you think it'll be better if these three different views are modularised as it would aid reusability and make this component look more compact.

Each possible view in this component has the **_div tag with className user-data_** **_and a p tag with className lead_**, I think it'll be better if you write this without having to repeat this tags in the different views.

Reply:
Thanks for your feedback. However, if you closely look at the component you'd see there's no point modularizing any part of this component. Inasmuch as all the blocks look similar I would like you to take a closer look at the lines below: 
- 12
- 26
- 58

Trying to break this component furthermore will only make it more difficult to understand or edit in the future. 

Thanks again. @Efosaok

Reply:
sorry I forgot to approve this earlier, I just felt it would be better modularizing this component, but if you insist it's better this way for this project, it's fine as well. as for the templating on those lines, correct me if I am wrong, but I do not see how modularizing it affects that part. the child component could receive the value as props.